csharp/src/Ice/Reference.cs

```diff
@@ -70,8 +70,157 @@ public static Reference Parse(string s, Communicator communicator, string? prope
+            bool? preferNonSecure = null;
+            RouterInfo? routerInfo = null;
+
+            // Override the defaults with the proxy properties if a property prefix is defined.
+            if (propertyPrefix != null && propertyPrefix.Length > 0)
+            {
+                // Warn about unknown properties.
+                if (communicator.WarnUnknownProperties)
+                {
+                    communicator.CheckForUnknownProperties(propertyPrefix);
+                }
+
+                cacheConnection = communicator.GetPropertyAsBool($"{propertyPrefix}.ConnectionCached");
+
+                string property = $"{propertyPrefix}.Context.";
+                context = communicator.GetProperties(forPrefix: property).
+                    ToDictionary(e => e.Key.Substring(property.Length), e => e.Value);
+                if (context.Count == 0)
+                {
+                    context = null;
+                }
+
+                property = $"{propertyPrefix}.EndpointSelection";
+                try
+                {
+                    endpointSelection =
+                        communicator.GetProperty(property) is string endpointSelectionStr ?
+                            Enum.Parse<EndpointSelectionType>(endpointSelectionStr) : (EndpointSelectionType?)null;
+                }
+                catch (FormatException ex)
+                {
+                    throw new InvalidConfigurationException($"cannot parse property `{property}'", ex);
+                }
+
+                property = $"{propertyPrefix}.InvocationTimeout";
+                invocationTimeout = communicator.GetPropertyAsInt(property);
+                if (invocationTimeout != null && invocationTimeout < 1 && invocationTimeout != -1)
+                {
+                    throw new InvalidConfigurationException(
+                        $"invalid value for property `{property}': `{invocationTimeout}'");
+                }
+
+                locatorInfo = communicator.GetLocatorInfo(
+                    communicator.GetPropertyAsProxy($"{propertyPrefix}.Locator", ILocatorPrx.Factory), encoding);
+
+                property = $"{propertyPrefix}.LocatorCacheTimeout";
+                locatorCacheTimeout = communicator.GetPropertyAsTimeSpan(property);
+                if (locatorCacheTimeout != null &&
+                    locatorCacheTimeout < TimeSpan.Zero &&
+                    locatorCacheTimeout != Timeout.InfiniteTimeSpan)
+                {
+                    throw new InvalidConfigurationException(
+                        $"invalid value for property `{property}': `{locatorCacheTimeout}'");
+                }
+
+                preferNonSecure = communicator.GetPropertyAsBool($"{propertyPrefix}.PreferNonSecure");
+
+                property = $"{propertyPrefix}.Router";
+                IRouterPrx? router = communicator.GetPropertyAsProxy(property, IRouterPrx.Factory);
+                if (router != null)
```

Comment:
We can inline router declaration here
```suggestion
                if (communicator.GetPropertyAsProxy(property, IRouterPrx.Factory) is IRouterPrx router)
```

Reply:
Fixed as suggested.